src/Command/App.php

```diff
@@ -43,18 +53,20 @@ protected function configure() : void
     protected function execute(InputInterface $input, OutputInterface $output) : int
     {
         $directory = $input->getArgument('directory');
-        $finder    = (new DocFinder($directory))->__invoke();
 
-        $output->writeln("\nFinding documentation files on <info>" . $directory . "</info>\n");
+        $documentFormatHandler = new MultipleDocumentationFormat(...array_map(
+            static fn (string $class) => new $class(),
+            $this->config['plugins']
+        ));
 
-        $documentFormatHandler = new MultipleDocumentationFormat(
-            new RstDocumentationFormat(),
-            new MarkdownDocumentationFormat()
-        );
+        $finder = (new DocFinder($directory, $documentFormatHandler->supportedExtensions()))
+            ->__invoke($this->config['exclude_dirs']);
```

Comment:
Similar to the above, the config is also an `@psalm-var array{exclude_dirs: list<non-empty-string>}`